File: PA01/perceptron.jl
#= perceptron.jl
perceptron function for perceptron.jl
PA02 (CS5000)
01170065 - Xavier Sánchez Díaz
=#

include("hardlimit.jl")

"""
Perceptron function.
Calculates the outputs of a perceptron given an input `p`
and based on a weight matrix `W` and a bias vector `b`,
using the `f` activation function.
"""
function perceptron(W, p, b, f)

    n = W * p + b
    out = map(f, n)

    return out
end

File: PA02/perceptron.m
# -- Mapping Function: perceptron(W, b, p, f)
#     Calculates the output(s) of a perceptron given an input p. The
#     perceptron is fully specified by a weight matrix W, a bias 
#     vector b and an activation function. 

function a = perceptron(W, b, p, f)
	n = W * p + b;
	a = arrayfun(f, n);
end